openhcl/virt_mshv_vtl/src/processor/mod.rs

```diff
@@ -675,11 +680,12 @@ impl<'p, T: Backing> Processor for UhProcessor<'p, T> {
                     }
                 }
 
-                if ready {
+                // TODO WHP GUEST VSM: This should be next_vtl
+                if T::halt_in_usermode(self, GuestVtl::Vtl0) {
+                    break Poll::Pending;
+                } else {
```

Comment:
So for x64 the halt gets handled here, but in e.g. snp it gets handled in run_vp_snp?

Reply:
For x64 with apic emulation it gets handled here, which is only the case in test scenarios (WHP). For all other scenarios we call set_halted on the runner and let the kernel manage it for us, as that's more efficient.

Reply:
I'm not entirely familiar with why we can't do that for WHP, I just tried to preserve existing behavior here.